src/hooks/useCostDetails.tsx

```diff
@@ -1,15 +1,33 @@
-import { AoGetCostDetailsParams } from '@ar.io/sdk';
+import {
+  AoARIORead,
+  AoARIOWrite,
+  AoGetCostDetailsParams,
+  CostDetailsResult,
+} from '@ar.io/sdk';
 import { useGlobalState } from '@src/state';
 import { useQuery } from '@tanstack/react-query';
 
-export function useCostDetails(params: AoGetCostDetailsParams) {
-  const [{ arioProcessId, arioContract }] = useGlobalState();
-  // we are verbose here to enable predictable keys. Passing in the entire params as a single object can have unpredictable side effects
-  return useQuery({
+export function buildCostDetailsQuery(
```

Comment:
Not sure why this function is separated out, would re-inline this code back into useCostDetails

Reply:
I pulled this out now because we are using the cost details api on the table to load prices and want to leverage the query cache - seperating the function like this ensures we use the same query args (key, function, staletime primaryily)